### PR TITLE
[change] api - Updated unsafe requests to use POST instead of GET

### PR DESCRIPTION
**Pull request by kristenmills**
### Motivation for changes:
So currently a bunch of unsafe requests are using `GET` instead of `POST`. Because all modern browsers support prefetching, it's not wise to use `GET` requests for things that do any sort of modification server side.  

### Detailed changes:
- Update `/logout`, `/server/reload`, `/server/shutdown`, `/database/vacuum`, `/database/cleanup`, and `/database/reset_plugin` to use `POST` instead of `GET`
- Update Web UI to account for these changed request methods.

### CODE REVIEW

**Comment by liiight** [comment]
This is a good idea. I wonder though if we should move the action to the payload of the request instead of being in the url itself. Something like `{"operation": "vacuum"}`. What do you think? 
I know POST without body is valid but I feel like using a body fits better.

**Comment by kristenmills** [comment]
I kind of like that too. We could combine the endpoints that way. Do you have a suggestion for the combined endpoint name? We could just do `POST /server` and `/database`? Maybe `/server/action` and `/database/action`?

**Comment by liiight** [comment]
I think i like `/server` and `/database` best

File: flexget/api/core/database.py [hunk 1]
```diff
@@ -17,53 +17,53 @@
...
+        data = request.json
+        operation = data['operation']
+        if operation == 'cleanup':
+            self.manager.db_cleanup(force=True)
+            msg = 'DB Cleanup finished'
+        elif operation == 'vacuum':
+            session.execute('VACUUM')
+            session.commit()
+            msg = 'DB VACUUM finished'
+        elif operation == 'list_plugins':
```
Comment (kristenmills) [inline flexget/api/core/database.py]: I think `list_plugins` should still be a GET request. It's idempotent and semantically that makes more sense.

Comment (liiight) [inline flexget/api/core/database.py]: True, i'll change it

**Comment by kristenmills** [comment]
Sorry I haven't gotten around to doing the other endpoint yet...I'll try to get to it soon.

**Comment by liiight** [comment]
You mean the server endpoint? I've been meaning to do that, been super busy...

**Comment by liiight** [comment]
I went with `/server/manage`

**Comment by kristenmills** [comment]
Alright I think this is good to go now. 👍

**Comment by liiight** [comment]
Actually I think we missed one, `/user/token` better be post as well

**Comment by kristenmills** [comment]
That one might more sense as a `put` but in general I agree.

**Comment by liiight** [comment]
Awesome, thank for this change!

**Pull request merged**